File: rules/java/weak-hashes.java
// Test fixtures for pqc-semgrep-rules / Java / classically-broken hashes.
import java.security.MessageDigest;

public class WeakHashesFixture {
    public static void main(String[] args) throws Exception {
        // ruleid: pqc.java.md5
        MessageDigest md5 = MessageDigest.getInstance("MD5");

        // ruleid: pqc.java.sha1
        MessageDigest sha1 = MessageDigest.getInstance("SHA-1");
        // ruleid: pqc.java.sha1
        MessageDigest sha1alt = MessageDigest.getInstance("SHA1");

        // ok: pqc.java.md5
        MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
    }
}

File: rules/python/weak-hashes.py
"""Test fixtures for pqc-semgrep-rules / Python / classically-broken hashes."""

import hashlib

from Crypto.Hash import MD5, SHA1


def vulnerable():
    # ---- MD5 ----
    # ruleid: pqc.python.md5
    hashlib.md5(b"x").hexdigest()
    # ruleid: pqc.python.md5
    hashlib.new("md5", b"x").hexdigest()
    # ruleid: pqc.python.md5
    MD5.new(b"x").hexdigest()

    # ---- SHA-1 ----
    # ruleid: pqc.python.sha1
    hashlib.sha1(b"y").hexdigest()
    # ruleid: pqc.python.sha1
    hashlib.new("sha1", b"y").hexdigest()
    # ruleid: pqc.python.sha1
    SHA1.new(b"y").hexdigest()


def safe():
    # ok: pqc.python.md5
    hashlib.sha256(b"safe").hexdigest()
    # ok: pqc.python.sha1
    hashlib.sha384(b"safe").hexdigest()
